filter_plugins/oo_filters.py

```diff
@@ -128,14 +128,14 @@ def oo_select_keys(data, keys):
                 returns [1, 3]
         """
 
-        if not isinstance(data, dict):
-            raise errors.AnsibleFilterError("|failed expects to filter on a dict")
+        if not isinstance(data, dict) and not isinstance(data, object):
+            raise errors.AnsibleFilterError("|failed expects to filter on a dict or object")
```

Comment:
After a bit of digging, we believe we better understand the issue here...  It looks like in Ansible 2.0+ passing `hostvars` without referencing a specific host passes a HostVars object into the filter rather than a dict. HostVars inherits from Mapping, so instead of comparing against object here, you should be able to just update the test as below:
```
if not isinstance(data, collections.Mapping):
```

@abutcher is doing some more testing around this as well.